packages/data-service/src/data-service.ts

```diff
@@ -727,6 +737,17 @@ export interface DataService {
    * is being emitted when this value changes.
    */
   getUpdatedSecrets(): Promise<Partial<ConnectionOptions>>;
+
+  /**
+   * Runs the update within a transactions, only
+   * modifying a subset of the documents matched by the filter.
+   * It returns a list of the changed documents, or a serverError.
+   */
+  previewUpdate(
```

Comment:
We can do this in this PR or a separate one, but I think we should probably take an options parameter containing an optional `signal` which would be an AbortSignal. So that if a new call comes in the UI can abort the old one. See how we can abort find and count and so on. There are also cancellableQueries elsewhere in the codebase.

Usually this should be fast, but I think even with limit if there's no index it could be scanning the whole collection if there aren't enough docs matching the filter. So in that case aborting it should be worth it.

(Mostly I think the abort signal will just be there for the calling code to ignore results coming in for aborted calls and not update the UI with stale results.)